database/prefixdb/db.go

```diff
@@ -236,23 +224,12 @@ func (db *Database) HealthCheck(ctx context.Context) (interface{}, error) {
 }
 
 // Return a copy of [key], prepended with this db's prefix.
-// The returned slice should be put back in the pool
-// when it's done being used.
-func (db *Database) prefix(key []byte) []byte {
-	// Get a []byte from the pool
-	prefixedKey := db.bufferPool.Get().([]byte)
+// The returned slice should be put back in the pool when it's done being used.
+func (db *Database) prefix(key []byte) *[]byte {
```

Comment:
wonder if we could get rid of the pointer here. Not sure if there is a way to return a byte slice and give back the slice internal memory to the pool one we are done using it.

Reply:
using `*[]byte` avoids memory allocations